lib/ddtrace/contrib/status_code_matcher.rb

```diff
@@ -0,0 +1,60 @@
+require 'set'
+
+module Datadog
+  module Contrib
+    # Contains methods helpful for tracing/annotating HTTP request libraries
+    class StatusCodeMatcher
+      REGEX_PARSER = /^\d{3}(?:-\d{3})?(?:,\d{3}(?:-\d{3})?)*$/
+
+      def initialize(range)
+        @error_response_range = range
+      end
+
+      def set_range
+        @datadog_set ||= begin
+          set = Set.new
+          handle_statuses.each do |statuses|
+            status = statuses.to_s.split('-')
+            if status.length == 1
+              set.add(Integer(status[0]))
+            elsif status.length == 2
+              min, max = status.minmax
+              Array(min..max).each do |i|
+                set.add(Integer(i))
+              end
+            end
+          end
+          set
+        end
+        @datadog_set
+      end
+
+      def to_s
+        @@error_response_range.to_s
+      end
+
+      private
+
+      def error_responses
+        return @error_response_range if @error_response_range.is_a?(String) && !@error_response_range.nil?
+        @error_response_range.join(',') if @error_response_range.is_a?(Array) && !@error_response_range.empty?
+      end
+
+      def handle_statuses
+        if error_responses
+          error_responses.gsub(/\s+/, '').split(',').select do |code|
+            if !code.to_s.match(REGEX_PARSER)
+              Datadog.logger.debug("Invalid config provided: #{code}. Must be formatted like '400-403,405,410-499'.")
+              next
+            else
+              true
+            end
+          end
+        else
+          Datadog.logger.debug('No valid config was provided for :error_responses - falling back to default.')
+          ['500-599'] # Rather than returning an empty array, we need to fallback to default config.
```

Comment:
`500-599` can be replaced with a sensibly named default constant within, `'ddtrace/ext/http', as discussed previously.